components/ai_chat/content/browser/ai_chat_tab_helper.cc

```diff
@@ -345,6 +389,58 @@ void AIChatTabHelper::CheckPDFA11yTree() {
   OnPDFA11yInfoLoaded();
 }
 
+void AIChatTabHelper::FetchSearchQuerySummary(
+    FetchSearchQuerySummaryCallback callback) {
+  if (!IsBraveSearchSERP(GetPageURL())) {
+    std::move(callback).Run(std::nullopt);
+    return;
+  }
+
+  auto internal_callback =
+      base::BindOnce(&AIChatTabHelper::OnSearchSummarizerKeyFetched,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback));
+  GetSearchSummarizerKey(web_contents(), std::move(internal_callback));
+}
+
+void AIChatTabHelper::OnSearchSummarizerKeyFetched(
+    FetchSearchQuerySummaryCallback callback,
+    const std::optional<std::string>& key) {
+  if (!key) {
+    std::move(callback).Run(std::nullopt);
+    return;
+  }
+
+  if (!api_request_helper_) {
+    api_request_helper_ =
```

Comment:
noting that this looks like a Tor bypass if this feature were available in Tor windows, but i don't think it is at the moment